plonky3/src/circuit_builder.rs

```diff
@@ -40,6 +40,8 @@ pub struct ConstraintSystem<T> {
     witness_columns: BTreeMap<PolyID, (usize, usize)>,
     // for each fixed column, the index of this column in the fixed columns
     fixed_columns: BTreeMap<PolyID, usize>,
+    // for each intermediate polynomial, the expression
+    intermediates: BTreeMap<u64, AlgebraicExpression<T>>,
```

Comment:
Hmm, should we use `PolyID` here? I know the polynomial type is redundant, but the semantics is clearer than and the overhead should be negligible.

Reply:
done